### PR TITLE
feat: add llama hugging face apis

### PR DESCRIPTION
**Pull request by mattzcarey**

### CODE REVIEW

File: genoss/entities/chat/chat_completion.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import time
+import uuid
+from typing import Any, Dict
+
+from genoss.entities.chat.messages import Message
+
+
+class ChatCompletion:
+    class Choice:
```
Comment (mattzcarey) [inline genoss/entities/chat/chat_completion.py:9]: enclosing classes makes more sense here

**Comment by StanGirard** [comment]
LGTM, what do you think @MaximeThoonsen ?

File: README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ cd ../../gpt4all-bindings/python
 pip3 install -e .
 ```
 
-7. Download it to your local machine from [here](https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin) and put it in the `genoss/model` directory as `genoss/model/ggml-gpt4all-j-v1.3-groovy.bin`
+7. Download it to your local machine from [here](https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin) and put it in the `local_models` directory as `local_models/ggml-gpt4all-j-v1.3-groovy.bin`
```
Comment (MaximeThoonsen) [inline README.md:75]: iI agree

File: genoss/api/embeddings_routes.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from ast import List
 from fastapi import APIRouter
-from genoss.model.gpt4all_llm import Gpt4AllLLM
+from genoss.model.llm.local.gpt4all import Gpt4AllLLM
```
Comment (MaximeThoonsen) [inline genoss/api/embeddings_routes.py:3]: Agree more clear

File: genoss/llm/fake_llm.py [hunk 4]
```diff
@@ -1,35 +1,37 @@
...
-        prompt_template = "Question from user: {question}?, Answer from bot:"
-        llm_chain = LLMChain(
-            llm=llm, prompt=PromptTemplate.from_template(prompt_template)
-        )
-        response_text = llm_chain(last_messages)
+
+        llm_chain = LLMChain(llm=llm, prompt=prompt_template)
+        response_text = llm_chain(question)
+
         print("###################")
```
Comment (StanGirard) [inline genoss/llm/fake_llm.py:28]: logger instead of that

File: genoss/llm/local/gpt4all.py [hunk 5]
```diff
@@ -2,36 +2,35 @@
...
         )
-        prompt_template = "Question from user: {question}?, Answer from bot:"
-        llm_chain = LLMChain(
-            llm=llm, prompt=PromptTemplate.from_template(prompt_template)
-        )
-        response_text = llm_chain(last_messages)
+
+        llm_chain = LLMChain(llm=llm, prompt=prompt_template)
+        response_text = llm_chain(question)
         print("###################")
```
Comment (StanGirard) [inline genoss/llm/local/gpt4all.py:28]: print

**Review by StanGirard — APPROVED** [review]

**Pull request merged**